include/micm/solver/rosenbrock.hpp

```diff
@@ -40,179 +39,96 @@
-    uint64_t number_of_steps_{};
-    /// @brief The number of accepted integrations
-    uint64_t accepted_{};
-    /// @brief The number of rejected integrations
-    uint64_t rejected_{};
-    /// @brief The number of LU decompositions
-    uint64_t decompositions_{};
-    /// @brief The number of linear solves
-    uint64_t solves_{};
-    /// @brief The number of times a singular matrix is detected. For now, this will always be zero as we assume the matrix
-    /// is never singular
-    uint64_t singular_{};
-
-    /// @brief Set all member variables to zero
-    void Reset();
-  };
-
   /// @brief An implementation of the Rosenbrock ODE solver
+  /// @tparam RatesPolicy Calculator of forcing and Jacobian terms
+  /// @tparam LinearSolverPolicy Linear solver
   ///
   /// The template parameter is the type of matrix to use
-  template<
-      template<class> class MatrixPolicy = Matrix,
-      class SparseMatrixPolicy = StandardSparseMatrix,
-      class LinearSolverPolicy = LinearSolver<SparseMatrixPolicy>,
-      class ProcessSetPolicy = ProcessSet>
+  template<class RatesPolicy, class LinearSolverPolicy>
   class RosenbrockSolver
   {
    public:
-    struct [[nodiscard]] SolverResult
-    {
-      /// @brief The new state computed by the solver
-      MatrixPolicy<double> result_{};
-      /// @brief The final state the solver was in
-      SolverState state_ = SolverState::NotYetCalled;
-      /// @brief A collection of runtime state for this call of the solver
-      SolverStats stats_{};
-      /// @brief The final time the solver iterated to
-      double final_time_{};
-    };
-
-    std::vector<Process> processes_;
     RosenbrockSolverParameters parameters_;
-    StateParameters state_parameters_;
-    ProcessSetPolicy process_set_;
     LinearSolverPolicy linear_solver_;
+    RatesPolicy rates_;
+    std::vector<std::size_t> jacobian_diagonal_elements_;
 
     static constexpr double DELTA_MIN = 1.0e-6;
 
-    /// @brief Default constructor
-    RosenbrockSolver();
+    /// @brief Solver parameters typename
+    using ParametersType = RosenbrockSolverParameters;
 
-    /// @brief Builds a Rosenbrock solver for the given system, processes, and solver parameters
-    /// @param system The chemical system to create the solver for
-    /// @param processes The collection of chemical processes that will be applied during solving
-    /// @param parameters Rosenbrock algorithm parameters
+    /// @brief Default constructor
+    /// @param parameters Solver parameters
+    /// @param linear_solver Linear solver
+    /// @param rates Rates calculator
+    /// @param jacobian Jacobian matrix
+    ///
+    /// Note: This constructor is not intended to be used directly. Instead, use the SolverBuilder to create a solver
     RosenbrockSolver(
-        const System& system,
-        const std::vector<Process>& processes,
-        const RosenbrockSolverParameters& parameters);
+        RosenbrockSolverParameters parameters,
```

Comment:
Looks good! Thank you for addressing the comments! Just one more comment but I don't need to review for this

Does it make sense to make it `const ref`? 
```suggestion
        const RosenbrockSolverParameters& parameters,
```